src/Aspire.Hosting/ApplicationModel/ContainerMountAnnotation.cs

```diff
@@ -14,11 +14,11 @@ public sealed class ContainerMountAnnotation : IResourceAnnotation
     /// <summary>
     /// Instantiates a mount annotation that specifies the source and target paths for a mount.
     /// </summary>
-    /// <param name="source">The source path of the mount.</param>
+    /// <param name="source">The source path if a bind mount or name if a volume. Can be <c>null</c> if the mount is an anonymous volume.</param>
     /// <param name="target">The target path of the mount.</param>
     /// <param name="type">The type of the mount.</param>
     /// <param name="isReadOnly">A value indicating whether the mount is read-only.</param>
-    public ContainerMountAnnotation(string source, string target, ContainerMountType type, bool isReadOnly)
+    public ContainerMountAnnotation(string? source, string target, ContainerMountType type, bool isReadOnly)
     {
         Source = source;
```

Comment:
Should there be validation that a source value is required if type is binding mount?

Reply:
Yeah that's a good idea. I'll add that validation here.

Reply:
Added, including tests.